### DISCUSSION TITLE
pytest - addopts - environment variables

### DISCUSSION THREAD

**Discussion by nmt1** [pytest-dev/pytest#7932] [General]
# description
I don't understand how I can change the addopts in  pytest.
I am not sure what is the approach, and can't find a clear/strict way within the pytest documentation. 

# content of pytest.ini
[pytest]
addopts = --strict

# content of conftest.py
exitfirst = os.getenv("env_exitfirst", "false")  # typically controlled via Jenkins

if exitfirst == "true":
**write --exitfirst to addopts?**

**Comment by symonk** [comment]
Hi @nmt1 why not just specify the -x | --exit-first flag in the pytest.ini directly via addopts = --strict -x.  I would also like to point you to the --max-fail flag as thats possibly something of interest to you also (if you are in and around `exitfirst` behaviour).

Now if you must use the env vars via jenkins as mentioned in your original comment, this is available:

```
export PYTEST_ADDOPTS="-x"
```

Please let me know if this is useful for you, if you have any other questions I can try my best to assist you.

**Reply by nmt1** [reply]
Thanks for the answer!

where can I write the cmd: export PYTEST_ADDOPTS="-x"
is it in a pytest_configure hook?

**Reply by symonk** [reply]
No it’s just an environment variable 😎
